src/werkzeug/wrappers/cors.py

```diff
@@ -0,0 +1,92 @@
+from ..http import dump_header
+from ..http import parse_set_header
+from ..utils import environ_property
+from ..utils import header_property
+
+
+class RequestCORSMixin(object):
+    """A mixin for :class:`BaseRequest` subclasses. Request objects that
+    mix this class in will automatically get descriptors for Cross
+    Origin Resource Sharing headers.
+
+    .. versionadded:: 1.0
+    """
+
+    access_control_origin = environ_property(
+        "HTTP_ORIGIN",
+        doc="""The origin header field indicates the host that the request
+        originated from.""",
+    )
+
+    access_control_request_headers = environ_property(
+        "HTTP_ACCESS_CONTROL_REQUEST_HEADERS",
+        load_func=parse_set_header,
+        doc="""The Access-Control-Request-Headers field is set on a preflight
+        request to indicate what headers will be sent on the cross
+        origin request. This allows the server to reply indicating
+        which headers are allowed.""",
+    )
+
+    access_control_request_method = environ_property(
+        "HTTP_ACCESS_CONTROL_REQUEST_METHOD",
+        doc="""The Access-Control-Request-Method field is set on a preflight
+        request to indicate which method will be used on the cross
+        origin request. This allows the server to reply indicating
+        which method is allowed.""",
+    )
+
+
+class ResponseCORSMixin(object):
+    """A mixin for :class:`BaseResponse` subclasses. Response objects that
+    mix this class in will automatically get descriptors for Cross
+    Origin Resource Sharing headers.
+
+    .. versionadded:: 1.0
+    """
+
+    allow_credentials = header_property(
```

Comment:
Unlike the `Request` properties, none of these are prefixed with `access_control_`.

Reply:
Yea, strange I did this, fixed.